pkg/network/usm/shared_libraries.go

```diff
@@ -145,23 +175,29 @@ type soRegistration struct {
 }
 
 // unregister return true if there are no more reference to this registration
-func (r *soRegistration) unregisterPath(pathID pathIdentifier) bool {
+func (r *soRegistration) unregisterPath(pathID pathIdentifier, soreg *soRegistry) bool {
 	currentUniqueProcessesCount := r.uniqueProcessesCount.Dec()
 	if currentUniqueProcessesCount > 0 {
 		return false
 	}
 	if currentUniqueProcessesCount < 0 {
 		log.Errorf("unregistered %+v too much (current counter %v)", pathID, currentUniqueProcessesCount)
+		soreg.libUnregisterErrors.Add(1)
```

Comment:
wondering if we want to increase that metric here

Reply:
it's an error path, so make sense
or you mean removed the counter as we have log.Errorf() already ?

Reply:
I'm not sure this is a correct location to put this counter
it can spam us, and mind blind us from a real issue

Reply:
I think we don't have the same perspective, I think telemetry is a great way to detect this kind of errors path that we will not check actively (need to create monitors on top of logs). It's just a counter tell us something was wrong and would need further investigation.

Reply:
I moved to the log.Warnf to Error